rti_connext_dds_cmake_module/cmake/rti_build_helper.cmake

```
@@ -529,18 +526,25 @@ function(rti_guess_connextdds_arch)
       if(CONNEXTDDS_ARCH)
         break()
       endif()
+
     endforeach()
-  endif()
 
-  if(NOT CONNEXTDDS_ARCH)
-    message(WARNING
-      "CONNEXTDDS_ARCH not specified. Please set "
-      "-DCONNEXTDDS_ARCH= to specify your RTI Connext DDS "
-      " architecture")
-  else()
-      message(STATUS "Selected CONNEXTDDS_ARCH: ${CONNEXTDDS_ARCH}")
-  endif()
+    if(NOT CONNEXTDDS_ARCH)
+      list(GET architecture_candidates 0 CONNEXTDDS_ARCH)
+      if(CONNEXTDDS_ARCH)
+        message(STATUS
+          "unsupported CMAKE_HOST_SYSTEM_NAME (${CMAKE_HOST_SYSTEM_NAME}) "
+          "or CMAKE_HOST_SYSTEM_PROCESSOR (${CMAKE_HOST_SYSTEM_PROCESSOR}). "
+          "Using architecture ${CONNEXTDDS_ARCH} anyway.")
+      else()
+        message(WARNING
+          "CONNEXTDDS_ARCH not specified. Please set "
+          "-DCONNEXTDDS_ARCH= to specify your RTI Connext DDS "
+          " architecture")
+      endif()
+    endif()
```

this message is missing after the fix?

```suggestion
    else()
      message(STATUS "Selected CONNEXTDDS_ARCH: ${CONNEXTDDS_ARCH}")
    endif()
```

---

ohh i forgot it, thanks!